specifyweb/frontend/js_src/lib/components/FormEditor/createView.ts

```diff
@@ -263,38 +263,63 @@ function createViewFromTemplate(
   const originalNames = filterArray(
     view.altViews.altViews.map(({ viewDef }) => viewDef)
   );
+
   const nameMapper = Object.fromEntries(
     originalNames.map((name) => [name, getUniqueDefinitionName(name, viewSets)])
   );
-  return {
-    ...viewSets,
-    views: [
-      ...viewSets.views,
-      {
-        ...view,
-        name,
-        altViews: {
-          ...view.altViews,
-          altViews: view.altViews.altViews.map((altView) => ({
-            ...altView,
-            viewDef:
-              typeof altView.viewDef === 'string'
-                ? nameMapper[altView.viewDef] ?? altView.viewDef
-                : altView.viewDef,
-          })),
-        },
-      },
-    ],
-    viewDefs: [
-      ...viewSets.viewDefs,
-      ...viewDefinitions.map((definition) => ({
-        ...definition,
+
+  const uniqueName = getUniqueDefinitionName(name, viewSets);
+
+  const updatedAltViews = view.altViews.altViews.map((altView) => ({
+    ...altView,
+    viewDef:
+      typeof altView.viewDef === 'string'
+        ? nameMapper[altView.viewDef] ?? altView.viewDef
+        : altView.viewDef,
+  }));
+
+  const updatedView = {
+    ...view,
+    name: uniqueName,
+    altViews: { ...view.altViews, altViews: updatedAltViews },
+  };
```

Comment:
nothing to change here, but just a FYI on why I often create giant pyramids of nested objects rather than break them down into separate variables:
- if I need readability, separate variables are usually better
- but, by nesting it all together in one large object, I get much better autocomplete and typescript error messages that highlight the exact place that caused the error (because typescript knows what is the expected type - in your case, you could make a typo on line 273-279, but the error would be highlighted all the way on line 321)
- if I want best of both worlds, I have to explicitly specify the variable type when creating a variable (on line 273 and 281 and 287) - that's what you get automatically when you define the object in the function `return` - typescript knows what the correct type should be

so it's a trade off. just one more thing to think about

cc @realVinayak @melton-jason